src/TextInput.js

```diff
@@ -1,167 +1,156 @@
-import React, { Component } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
-
-    this.id = props.id || idgen();
-
-    if (props.password) {
-      this.id = `password${idgen()}`;
+const TextInput = ({
+  value,
+  children,
+  s,
+  m,
+  l,
+  xl,
+  disabled,
+  noLayout,
+  placeholder,
+  icon,
+  label,
+  inputClassName,
+  success,
+  error,
+  password,
+  email,
+  validate,
+  defaultValue,
+  type,
+  ...props
+}) => {
+  const newId = id => {
```

Comment:
Since this is a function, you could add a prefix to make more explicit that it is a function. Ex.:
- `getNewId`
- `createNewId`
- `generateNewId`
...

Reply:
yes, I would vote to move this to defaultProps, `id: idgen()`

Reply:
@alextrastero , you say move the function to the idgen.js file or use the idgen() function ?
because if is the second option, the problem is to set again the ID in this situations:
`    if (password) {
      id = `password${idgen()}`;
    }

    if (email) {
      id = `email${idgen()}`;
    }`

Reply:
Sorry about the delay